### PR TITLE
[13.0][FIX] account_invoice_ubl_email_attachment: attach only the most recently generated UBL file

### PR DESCRIPTION
**Pull request by thomaspaulb**
Forward port of https://github.com/OCA/edi/pull/287

### CODE REVIEW

**Review by astirpe — APPROVED** [review]

**Review by StefanRijnhart — APPROVED** [review]

File: account_invoice_ubl_email_attachment/models/mail_template.py [hunk 1]
```diff
@@ -26,7 +26,9 @@ def generate_email(self, res_ids, fields=None):
                     ("res_model", "=", "account.move"),
                     ("res_id", "=", res_id),
                     ("name", "=", ubl_filename),
-                ]
+                ],
+                order="create_date desc",
+                limit=1,
```
Comment (etobella) [inline account_invoice_ubl_email_attachment/models/mail_template.py:31]: If you are adding the limit, then the `if len(ubl_attachments) == 1` has no sense.

Can you explain why the limit=1 is important?

Comment (thomaspaulb) [reply account_invoice_ubl_email_attachment/models/mail_template.py:31]: See https://github.com/OCA/edi/pull/287, if we dont have the limit then all the previous versions of the UBL will also be attached, which makes no sense. I agree the code can be shortened though, will push a fixup.

**Comment by etobella** [comment]
@thomaspaulb can you squash your commits and then proceed with merge :wink:

**Comment by thomaspaulb** [comment]
@etobella Done.

**Pull request merged**